File: src/webScraping.py
import newsAPI
import json
import spacy

nlp = spacy.load("en_core_web_sm")

class WebScraping:
    def __init__(self):
        self.r = newsAPI.getTopStories().json()

    def getTitlesAndDesriptions(self):
        articleTitles = []
        articleDesc = []
        for article in self.r['articles']:
            articleTitles.append(article['title'])
            articleDesc.append(article['description'])
        return articleTitles, articleDesc

    def getTitles(self):
        articleTitles = []
        for article in self.r['articles']:
            articleTitles.append(article['title'])
        return articleTitles

    def getDescriptions(self):
        articleDesc = []
        for article in self.r['articles']:
            articleDesc.append(article['description'])
        return articleDesc

if __name__ == "__main__":
    newInstance = WebScraping()
    titles, desc = newInstance.getTitlesAndDesriptions()
    text = titles[1]
    doc = nlp(text)

    for entity in doc.ents:
        print(entity.text, entity.label_)
    

File: src/javascript/getArticle/webScraping.js
var newTitle = require('./getNewTitle')
const fetch = require('node-fetch');

async function getEverything(){
    fetch('https://newsapi.org/v2/top-headlines?country=us&apiKey=45e32894443047b8bde86b7d9ce8720d')
        .then(res => res.json())
        .then((json) => {
            var articleTitles = []
            var articleDesc = []
            for (var article in json.articles){
                article = json.articles[article]
                articleTitles.push(article.title)
                articleDesc.push(article.description)
            }
            console.log(articleTitles)
            return newTitle([articleTitles, articleDesc])
        });
} 


module.exports = getEverything